tests/unit/components/gallery/HdGalleryPlaceholder.spec.js

```diff
@@ -0,0 +1,31 @@
+import _ from 'lodash';
+import { wrapperFactoryBuilder } from 'tests/unit/helpers';
+import HdGalleryPlaceholder from '@/components/gallery/HdGalleryPlaceholder.vue';
+
+describe('HdGalleryPlaceholder', () => {
+  const build = (props = {}) => {
+    const wrapper = wrapperFactoryBuilder(HdGalleryPlaceholder, {
+      props: _.assign({
+        icon: '',
+        text: '',
+      }, props),
```

Comment:
The idea from the factory is that you initialize it with the defaults and you can overwrite it in your tests so no need to do this `assign` again.

For example: https://github.com/homeday-de/homeday-blocks/blob/develop/tests/unit/components/form/HdInputFormatter.spec.js#L4-L13 and https://github.com/homeday-de/homeday-blocks/blob/develop/tests/unit/components/form/HdInputFormatter.spec.js#L124-L135

Reply:
The same applies to all other tests. Your factory should be able to create a working wrapper at the end without any extra params. In the tests you can overwrite those defaults if needed.

Reply:
You could have the factory outside the builder, like

```javascript
const wrapperFactory = wrapperFactoryBuilder(HdGalleryPlaceholder, {
      props: {
        icon: '',
        text: '',
      });
```

And use it inside your `build` something like that.

```javascript
const build = (...params) => {
    const wrapper = wrapperFactory(...params);
...
```

The idea is to avoid repeating `props` and everything you need to forward to the factory as it has been already defined somewhere else.

Reply:
Or you could also extend the factory to have something like your `getters` (I will call them getters), could work as well but I'm not sure how it would look like exactly (maybe another PR as it is going to another scope already).

Reply:
I see, that makes sense.

In the factory function, we are just overriding the values from props: https://github.com/homeday-de/homeday-blocks/blob/364e324ca88133efaf5438a0f84054b3ed7c7ae8/tests/unit/helpers.js#L51 right?

Before I apply the changes, do you think I should refactor the factory to use the `_.assign`?

I'm questioning this because maybe the prop can be an object. 
And if we just concat the values with spread, we may lose the default values.

e.g:

``` javascript
function factoryBuilder(defaultProps) {
    return (overrideProps) => ({ ...defaultProps, ...overrideProps })
}

const userFactory = factoryBuilder({
    name: 'Daniel',
    item: {
        title: 'Hello World',
        children: []
    }
})

console.log(userFactory())
// { name: 'Daniel', item: { title: 'Hello World', children: [] } }

console.log(userFactory({
    item: {
        children: [1, 2, 3]
    }
}))
// { name: 'Daniel', item: { children: [ 1, 2, 3 ] } }
```

We've lost the default values, so we must explicitly concat the default values from the object

``` javascript
const defaultValues = {
    name: 'Daniel',
    item: {
        title: 'Hello World',
        children: []
    }
}
const userFactory = factoryBuilder(defaultValues)

console.log(userFactory({
    item: {
        ...defaultValues.item
        children: [1, 2, 3]
    }
}))
// { name: 'Daniel', item: { title: 'Hello World', children: [1, 2, 3] } }
```

If we use the `_.assign` that operation is done automatically.

``` javascript
function factoryBuilder(defaultProps) {
    return (overrideProps) => _.assign({ ...defaultProps, ...overrideProps })
}

// ......

console.log(userFactory({
    item: {
        children: [1, 2, 3]
    }
}))
// { name: 'Daniel', item: { title: 'Hello World', children: [1, 2, 3] } }
```

Do you think is relevant?

Reply:
> I see, that makes sense.
> 
> Thanks for the comments man 😄 I'll apply your suggestions.
> 
> In the factory function, we are just overriding the values from props:
> 
> https://github.com/homeday-de/homeday-blocks/blob/364e324ca88133efaf5438a0f84054b3ed7c7ae8/tests/unit/helpers.js#L51
> 
> right?
> Before I apply the changes, do you think I should refactor the factory to use the `_.assign`?
> 
> I'm questioning this because maybe the prop can be an object.
> And if we just concat the values with spread, we may lose the default values.

The "object" case is ignored for props, we have the deepmerge approach in realtor-app (https://github.com/homeday-de/realtor-app/blob/develop/tests/unit/utils.js#L67). I think I tried the `assign` from lodash but had issues using Jest mocks, not 100% sure though hehehe

> Do you think is relevant?

Answering it: yes, I think it is relevant hehehe

Reply:
Awesome!! I will refactor that part then, 1 min.

And I think I will just use the `deepmerge` dependency since we had problem with the `_.assign`

I'm used to use `ramda` instead of `lodash`, they have a https://ramdajs.com/docs/#mergeDeepRight function

Reply:
What do you think?

- 407973d
- 267dfb2
- ad59b5e

Reply:
Rambda seems a nice option. Never used it, just heard. Is it lighter than Lodash? I mean, if we import each function separately?

Reply:
Well, Ramda is a bit lighter, but it is a functional utility library.

Lodash has its functional module that already solves a lot of problems.
But IMHO I think Ramda it's nicer.

But they are just utility libraries xD